api/src/main/java/org/openmrs/module/registrationcore/api/mpi/openempi/OpenEmpiPatientFetcher.java

```diff
@@ -36,15 +38,20 @@
     private RegistrationCoreProperties coreProperties;
 
     @Override
-    public Patient fetchMpiPatient(String patientId) {
-        OpenEmpiPatientResult mpiPatient = queryExecutor.getPatientById(authenticator.getToken(), patientId);
+    public MpiPatient fetchMpiPatient(PatientIdentifier patientId) {
+        OpenEmpiPatientResult mpiPatient = queryExecutor.getPatientById(authenticator.getToken(), patientId.getIdentifier());
 
         return buildPatient(mpiPatient);
     }
 
-    private Patient buildPatient(OpenEmpiPatientResult mpiPatient) {
-        patientBuilder.setPatient(new Patient());
-        Patient patient = patientBuilder.buildPatient(mpiPatient);
+    @Override
+    public MpiPatient fetchMpiPatient(String patientId, String identifierTypeUuid) {
+        throw new NotImplementedException("Method fetchMpiPatient for OpenEmpiPatientFetcher is not implemented yet");
```

Comment:
Why not throw APIException?

Reply:
The error message can be used to portray the information, the downside to throwing these other exception types is that you code doesn't benefit from the generic framework code that handles APIExceptions, may be it's not a big deal since there are areas where we throw NPE or IllegalArgumentException but these are standard while NotImplementedException is not standard, it's coming from another library.

Reply:
ok got it, changed to API Exception